set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java

```diff
@@ -61,13 +61,22 @@ private UserSetQuery buildSearchQuery(Map<String, String> searchCriteria, String
 	
 	addTitleLangCriterion(searchCriteria, searchQuery);
 	
-	searchQuery.setSortCriteria(toArray(sort));	
+	addSortCriterion(searchCriteria, searchQuery, sort);
+	
 	searchQuery.setPageSize(pageSize);
 	searchQuery.setPageNr(page);
 	
 	return searchQuery;
     }
 
+    private void addSortCriterion(Map<String, String> searchCriteria, UserSetQuery searchQuery, String sort) throws ParamValidationException {
+      if (sort!=null && sort.contains(WebUserSetFields.TEXT_SCORE_SORT) && !searchCriteria.containsKey(WebUserSetFields.TEXT)) {
+          throw new ParamValidationException(I18nConstants.INVALID_PARAM_VALUE, I18nConstants.INVALID_PARAM_VALUE,
```

Comment:
this verification needs to be made in the parameter verification phase in the controller

Reply:
Since we now throw an exception if "score asc" order is in the input, I keep the processing of the exceptions here. Otherwise the processing would need to be added to the parseCriterion method, which would require to call the toArray method twice, etc. Please see also that the exceptions are also handled in other add* methods in the buildSearchQuery method. It poses a certain overhead, but a very minimalistic one, I would say.